tests/unit/DocBlock/Tags/ReturnTest.php

```diff
@@ -136,14 +137,49 @@ public function testFactoryMethod()
 
         $fixture = Return_::create('string My Description', $resolver, $descriptionFactory, $context);
 
-        $this->assertSame('string My Description', (string)$fixture);
+        $this->assertSame('string My Description', (string) $fixture);
         $this->assertEquals($type, $fixture->getType());
         $this->assertSame($description, $fixture->getDescription());
     }
 
     /**
+     * This test checks whether a braces in a Type are allowed.
+     *
+     * The advent of generics poses a few issues, one of them is that spaces can now be part of a type. In the past we
+     * could purely rely on spaces to split the individual parts of the body of a tag; but when there is a type in play
+     * we now need to check for braces.
+     *
+     * This test tests whether an error occurs demonstrating that the braces were taken into account; this test is still
+     * expected to produce an exception because the TypeResolver does not support generics.
+     *
      * @covers ::create
-     * @expectedException \InvalidArgumentException
+     * @uses \phpDocumentor\Reflection\DocBlock\Tags\Return_::<public>
+     * @uses \phpDocumentor\Reflection\DocBlock\DescriptionFactory
+     * @uses \phpDocumentor\Reflection\TypeResolver
+     * @uses \phpDocumentor\Reflection\DocBlock\Description
+     * @uses \phpDocumentor\Reflection\Types\String_
+     * @uses \phpDocumentor\Reflection\Types\Context
+     */
+    public function testFactoryMethodWithGenericWithSpace()
+    {
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('"\array😁<string,😁 string>" is not a valid Fqsen.');
```

Comment:
Surely the 2nd emoji should be to the right of the space character, not the left?

Reply:
Added another one to the right for fun :)